x/auth/middleware/auth_ext.go

```diff
@@ -0,0 +1,75 @@
+package middleware
+
+import (
+	"context"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/tx"
+)
+
+// ExtensionOptionChecker is a function that returns true if the extension option is accepted.
+type ExtensionOptionChecker func(*codectypes.Any) bool
+
+type HasAuthExtensionOptionsTx interface {
+	AuthExtensionOptions() []*codectypes.Any
+}
+
+type checkAuthExtensionOptionsTxHandler struct {
+	next    tx.Handler
+	checker ExtensionOptionChecker
+}
+
+// NewAuthExtensionOptionsMiddleware creates a new checkAuthExtensionOptionsMiddleware.
+// NewAuthExtensionOptionsMiddleware is a middleware that checks all auth_info extension
+// options pass the checker.
+func NewAuthExtensionOptionsMiddleware(checker ExtensionOptionChecker) tx.Middleware {
+	return func(next tx.Handler) tx.Handler {
+		return checkAuthExtensionOptionsTxHandler{
+			next,
+			checker,
+		}
+	}
+}
+
+var _ tx.Handler = checkAuthExtensionOptionsTxHandler{}
+
+func checkAuthExtOpts(tx sdk.Tx, checker ExtensionOptionChecker) error {
+	if hasExtOptsTx, ok := tx.(HasAuthExtensionOptionsTx); ok {
+		for _, opt := range hasExtOptsTx.AuthExtensionOptions() {
```

Comment:
By default, a transaction should not have Auth / Fee extensions set (it should be nil) and we should return error if it's set.

Reply:
I made the HandlerOptions accept a FeeMarket, and there's a default StaticFeeMarket implementation, that will return false for this interface.